drivers/sensor/pni/rm3100/rm3100_stream.c

```diff
@@ -32,7 +32,7 @@ static void rm3100_complete_result(struct rtio *ctx, const struct rtio_sqe *sqe,
 	if (!edata->header.events.drdy) {
 		LOG_ERR("Status register does not have DRDY bit set: 0x%02x",
 			edata->header.status);
-	} else if (data->stream.settings.opt == SENSOR_STREAM_DATA_INCLUDE) {
+	} else if (data->stream.settings.opt != SENSOR_STREAM_DATA_DROP) {
```

Comment:
Not a fan of Include/NOP/Drop but, isn't this effectively _including the data_ for `SENSOR_STREAM_DATA_NOP`? IIUC - the rationale behind a NOP is to capture the occurrence of an event without its data.

https://github.com/zephyrproject-rtos/zephyr/blob/67f246477f93ac8e7bd4b953a9f47d4f5f5b4ef5/drivers/sensor/pni/rm3100/rm3100_decoder.c#L196-L198

Reply:
Ok - Just to clarify: I think the "consume later" part implies there's an internal buffer within the sensor that allows the sample to be sitting there and consumed afterwards, which the RM3100 does not have (but the ICM45686 does).
Please note how `SENSOR_DT_READ_IODEV()`  does _not_ have a trigger (INCLUDE/NOP/DROP) option. Instead the secondary argument is a channel index, that would allow you to access multiple samples of the same channel.

Maybe to avoid confusion it would be better just to guard this sensor against _SENSOR_STREAM_DATA_NOP_, since it either grabs or discards the sample.